components/upload-progress-fullpage.tsx

```diff
@@ -74,17 +77,58 @@ const UploadProgressFullpage: React.FC<Props> = ({ file }) => {
     }
   }, [upload]);
 
+
+  const startFileValidation = (_file: File): Promise<void> => {
+    return new Promise((resolve, reject) => {
+      // Attempt to load the file as a video element and inspect its duration
+      // metadata. This is not an authoritative check of video duration, but
+      // rather intended to serve as just a simple and fast sanity check.
+      if (!_file.type.includes("video")) {
+        console.warn(`file type (${_file.type}) does not look like video!`);
+        resolve();
+      }
+
+      const video = document.createElement('video');
+      video.preload = 'metadata';
+      video.onloadedmetadata = function() {
+        URL.revokeObjectURL(video.src);
+        if (video.duration !== Infinity && video.duration > MAX_VIDEO_DURATION_MIN*60) {
+          const dur = Math.round(video.duration * 100) / 100;
+          reject(`file duration (${dur.toString()}s) exceeds allowed maximum (${MAX_VIDEO_DURATION_MIN}min)!`);
+        }
+        resolve();
+      };
+      video.onerror = function() {
+        // The file has a video MIME type, but we were unable to load its
+        // metadata for some reason.
+        console.warn("failed to load video file metadata for validation!");
+        URL.revokeObjectURL(video.src);
+        resolve();
+      };
+      video.src = URL.createObjectURL(file);
+    });
+  };
+
   useEffect(() => {
-    if (file) {
-      startUpload(file);
+    if (!file) {
+      return;
     }
+
+    startFileValidation(file).then(() => {
+      startUpload(file);
+    }).catch((error => {
+      setErrorMessage(error);
+    }));
   }, [file]);
 
   return (
     <Layout centered spinningLogo>
       {
         (errorMessage || error)
-          ? <div><h1>{(error && 'Error fetching API') || errorMessage}</h1></div>
+          ? <div className="errorMsg"><h1>Oops there was a problem uploading your file!</h1>
```

Comment:
is this class `errorMsg` used anywhere?

I think the intention for `errorMessage` here is intended to be user-facing error messages. For human-friendly errors that give more of an idea about what happened instead of a generic "Oops"

Reply:
Oops that's a holdover from when I was trying something else, I'll remove the class. My idea was to have a generic "big" error message and then details below (in `errorMessage`), similar to what's done in https://github.com/muxinc/stream.new/blob/main/pages/assets/%5Bid%5D.tsx#L52

Reply:
Oh whops, I skimmed over that. Good shout!  I'll test it out